src/Microsoft.ML.LightGbm/LightGbmBinaryTrainer.cs

```diff
@@ -116,15 +189,24 @@ internal LightGbmBinaryClassificationTrainer(IHostEnvironment env, Options optio
             int? numberOfLeaves = null,
             int? minimumExampleCountPerLeaf = null,
             double? learningRate = null,
-            int numberOfIterations = Trainers.LightGbm.Options.Defaults.NumberOfIterations)
-            : base(env, LoadNameValue, TrainerUtils.MakeBoolScalarLabel(labelColumnName), featureColumnName, exampleWeightColumnName, null, numberOfLeaves, minimumExampleCountPerLeaf, learningRate, numberOfIterations)
+            int numberOfIterations = Defaults.NumberOfIterations)
+            : this(env,
+                  new Options() {
+                      LabelColumnName = labelColumnName,
+                      FeatureColumnName = featureColumnName,
+                      ExampleWeightColumnName = exampleWeightColumnName,
+                      NumberOfLeaves = numberOfLeaves,
+                      MinimumExampleCountPerLeaf = minimumExampleCountPerLeaf,
+                      LearningRate = learningRate,
+                      NumberOfIterations = numberOfIterations
+                  })
         {
         }
 
         private protected override CalibratedModelParametersBase<LightGbmBinaryModelParameters, PlattCalibrator> CreatePredictor()
         {
             Host.Check(TrainedEnsemble != null, "The predictor cannot be created before training is complete");
-            var innerArgs = LightGbmInterfaceUtils.JoinParameters(Options);
+            var innerArgs = LightGbmInterfaceUtils.JoinParameters(base.GbmOptions);
```

Comment:
>base [](start = 66, length = 4)

Does `base` necessary?

Reply:
no that comes from the VS renaming....I will remove.

---
In reply to: [266665665](https://github.com/dotnet/machinelearning/pull/2948#discussion_r266665665) [](ancestors = 266665665)